### PR TITLE
Create-Signals.py first iteration

### PR DESCRIPTION
**Pull request by JohannesAhlstrom**

### CODE REVIEW

**Review by PeterAngell — APPROVED** [review]
looks ok

File: generators/signals.py [hunk 1]
```diff
@@ -1 +1,61 @@
...
+# try:
+#     if sys.argv[4] in modes:
+#         current_mode = sys.argv[4]
+# except Exception:
+#     print("Could not find mode")
+# try:
+#     if sys.argv[2] in nodes:
+#         current_node = sys.argv[2]
+# except Exception:
+#     print("Could not find node")
```
Comment (Merculafil) [inline generators/signals.py:27]: Shoud this be commented out?

File: generators/signals.py [hunk 2]
```diff
@@ -1 +1,61 @@
...
+
+if len(getter_signal_list)>1:
+    textfile.write("\n"+' '*45 +"GETTERS\n\n")
+for signal in getter_signal_list:
+    if 'range' in signal:
+        textfile.write(f"{signal['name']}"+ ' '* (40-(len(signal['name']))) + f"{signal['type']}" + ' '* (15-(len(signal['type']))) + f"{signal['range']}" + ' '* (40-(len(repr((signal['range']))))) + f"{signal['comment']}"+ f"\n")
+    else:
+        textfile.write(f"{signal['name']}"+ ' '* (40-(len(signal['name']))) + f"{signal['type']}" + ' '* (15-(len(signal['type'])))+ f"{data['defines'][signal['values']]}" + ' '*(40-(len(repr(data['defines'][signal['values']])))) + f"{signal['comment']}" + f"\n")
+
+textfile.close()
```
Comment (Merculafil) [inline generators/signals.py:61]: I can see that we close 'textfile', but do we close 'file'?

File: generators/signals.py [hunk 3]
```diff
@@ -1 +1,61 @@
...
+    else:
+        textfile.write(f"{signal['name']}"+ ' '* (40-(len(signal['name']))) + f"{signal['type']}" + ' '* (15-(len(signal['type'])))+ f"{data['defines'][signal['values']]}" + ' '*(40-(len(repr(data['defines'][signal['values']])))) + f"{signal['comment']}" + f"\n")
+
+if len(getter_signal_list)>1:
+    textfile.write("\n"+' '*45 +"GETTERS\n\n")
+for signal in getter_signal_list:
+    if 'range' in signal:
+        textfile.write(f"{signal['name']}"+ ' '* (40-(len(signal['name']))) + f"{signal['type']}" + ' '* (15-(len(signal['type']))) + f"{signal['range']}" + ' '* (40-(len(repr((signal['range']))))) + f"{signal['comment']}"+ f"\n")
+    else:
+        textfile.write(f"{signal['name']}"+ ' '* (40-(len(signal['name']))) + f"{signal['type']}" + ' '* (15-(len(signal['type'])))+ f"{data['defines'][signal['values']]}" + ' '*(40-(len(repr(data['defines'][signal['values']])))) + f"{signal['comment']}" + f"\n")
```
Comment (Merculafil) [inline generators/signals.py:59]: Maybe this should be input/output instead of getter/setter?

**Review by Merculafil — COMMENTED** [review]
Look over my comments :)

File: generators/signals.py (hunk 1 above)
Comment (JohannesAhlstrom) [reply generators/signals.py:27]: For now, yes

File: generators/signals.py (hunk 3 above)
Comment (JohannesAhlstrom) [reply generators/signals.py:59]: Not according to Pontus, the input/output part is not really needed it seems, But I dont want to remove it yet.

File: generators/signals.py (hunk 2 above)
Comment (JohannesAhlstrom) [reply generators/signals.py:61]: Nope,  I'll check that

Comment (JohannesAhlstrom) [reply generators/signals.py:62]: It's now being closed.

**Review by Merculafil — APPROVED** [review]

**Review by Merculafil — APPROVED** [review]

**Pull request merged**